src/visitor.rs

```diff
@@ -145,20 +145,24 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
 
         match item.node {
             ast::Item_::ItemUse(ref vp) => {
+                self.format_missing_with_indent(item.span.lo);
                 match vp.node {
                     ast::ViewPath_::ViewPathList(ref path, ref path_list) => {
-                        self.format_missing(item.span.lo);
-                        let new_str = self.rewrite_use_list(path,
-                                                            path_list,
-                                                            item.vis,
-                                                            vp.span);
-                        self.changes.push_str_span(item.span, &new_str);
+                        let block_indent = self.block_indent;
+                        let budget = IDEAL_WIDTH - block_indent;
+                        if let Some(new_str) = self.rewrite_use_list(block_indent,
+                                                                     budget,
+                                                                     path,
+                                                                     path_list,
+                                                                     item.vis) {
+                            self.changes.push_str_span(item.span, &new_str);
+                        }
                         self.last_pos = item.span.hi;
```

Comment:
I think you should only set last_pos if you push the string, otherwise we'll miss the import.